src/app/query/stacks/balance/stacks-ft-balances.hooks.ts

```diff
@@ -1,140 +1,38 @@
-  const { data: initializedAssetBalances = [] } = useStacksFungibleTokenAssetBalances(address);
-  const priceAsMarketData = useAlexCurrencyPriceAsMarketData();
-
-  const ftAssetsMetadata = useGetFungibleTokenMetadataListQuery(
-    initializedAssetBalances.map(assetBalance =>
-      formatContractId(assetBalance.asset.contractAddress, assetBalance.asset.contractName)
-    )
-  );
-
-  return useMemo(
-    () =>
-      initializedAssetBalances.map((assetBalance, i) => {
-        const metadata = ftAssetsMetadata[i].data;
-        if (!(metadata && isFtAsset(metadata))) return assetBalance;
-        return addQueriedMetadataToInitializedStacksFungibleTokenAssetBalance(
-          assetBalance,
-          metadata,
-          priceAsMarketData(
-            formatContractId(assetBalance.asset.contractAddress, assetBalance.asset.contractName),
-            metadata.symbol
-          )
-        );
-      }),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [initializedAssetBalances]
-  );
-}
-
-export function useStacksFungibleTokenAssetBalance(contractId: string) {
-  const toast = useToast();
-  const account = useCurrentStacksAccount();
-  const navigate = useNavigate();
-  const assetBalances = useStacksFungibleTokenAssetBalancesWithMetadata(account?.address ?? '');
-
-  return useMemo(() => {
-    const balance = assetBalances.find(assetBalance =>
-      assetBalance.asset.contractId.includes(contractId)
-    );
-    if (!balance) {
-      toast.error('Unable to find balance by contract id');
-      navigate('..');
-    }
-    return balance ?? createStacksFtCryptoAssetBalanceTypeWrapper(new BigNumber(0), contractId);
-  }, [assetBalances, contractId, navigate, toast]);
-}
-
-export function useTransferableStacksFungibleTokenAssetBalances(
-  address: string
-): StacksFungibleTokenAssetBalance[] {
-  const assetBalances = useStacksFungibleTokenAssetBalancesWithMetadata(address);
-  return useMemo(
-    () => assetBalances.filter(assetBalance => assetBalance.asset.canTransfer),
-    [assetBalances]
-  );
-}
-
-function filterStacksFungibleTokens(
-  assetBalances: StacksFungibleTokenAssetBalance[],
-  swapAssets: SwapAsset[],
-  filter: StacksFtTokensFilter
-) {
-  if (filter === 'supported') {
-    return assetBalances.filter(assetBalance =>
-      swapAssets.some(swapAsset => swapAsset.principal.includes(assetBalance.asset.contractAddress))
-    );
-  }
-
-  if (filter === 'unsupported') {
-    return assetBalances.filter(
-      assetBalance =>
-        !swapAssets.some(swapAsset =>
-          swapAsset.principal.includes(assetBalance.asset.contractAddress)
-        )
-    );
-  }
+// TODO: Asset refactor: remove if determined unnecessary
+// ts-unused-exports:disable-next-line
+export function useSip010CryptoAssetBalances(address: string) {
+  const { data: tokens = {} } = useStacksAccountBalanceFungibleTokens(address);
+  const tokenMetadata = useStacksAccountFungibleTokenMetadata(tokens);
```

Comment:
Couldn't this better consume `useStacksAccountBalanceQuery` directly so you can use the `select` to modify the data?

Reply:
Let me try. 👍

Reply:
I'm struggling to find a good solution here with dependent react-query hooks and using `select`. 🤔 I'm not finding any refs that suggest there is way to do this? I don't feel like these examples are diff from what I've already proposed? https://tanstack.com/query/latest/docs/framework/react/guides/dependent-queries

Maybe we could not use react-query for the token metadata fetch?

Reply:
This seems to work: 

```ts
export function useSip010CryptoAssetBalances(address: string) {
  const client = useStacksClient();
  const limiter = useHiroApiRateLimiter();

  return useStacksAccountBalanceQuery(address, {
    select: resp => {
      const contractIds = Object.keys(resp.fungible_tokens).map(key =>
        pullContractIdFromIdentity(key)
      );
      const tokenMetadata = contractIds.map(principal =>
        fetchFungibleTokenMetadata(client, limiter)(principal)
      );
      return Object.entries(resp.fungible_tokens)
        .map(([_, value], i) => createSip010CryptoAssetBalance(value.balance, tokenMetadata[i]))
        .filter(isDefined);
    },
  });
}
```